components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/micro/integrator/inbound/endpoint/protocol/grpc/GRPCInjectHandler.java

```diff
@@ -0,0 +1,203 @@
+        if (sequenceName != null) {
+            if (log.isDebugEnabled()) {
+                log.debug(sequenceName + " sequence, received via gRPC headers.");
+            }
+            seq = (SequenceMediator) synapseEnvironment.getSynapseConfiguration().getSequence(sequenceName);
+        } else {
+            if (injectingSeq == null || injectingSeq.isEmpty()) {
+                log.error("Sequence name is not specified in inbound endpoint or empty.");
+                return;
+            }
+            if (log.isDebugEnabled()) {
+                log.debug(injectingSeq + " sequence, received via the inbound endpoint.");
+            }
+            seq = (SequenceMediator) synapseEnvironment.getSynapseConfiguration().getSequence(injectingSeq);
+        }
+        msgCtx.setProperty(SynapseConstants.IS_INBOUND, true);
+        //validating the sequence
+        if (seq != null) {
+            if (!seq.isInitialized()) {
+                seq.init(synapseEnvironment);
+            }
+            seq.setErrorHandler(onErrorSeq);
+            if (log.isDebugEnabled()) {
+                log.debug("injecting received gRPC message to sequence : " + injectingSeq);
+            }
+            if (!synapseEnvironment.injectInbound(msgCtx, seq, this.sequential)) {
+                return;
+            }
+        } else {
+            log.error("Sequence: " + injectingSeq + " not found");
+        }
+        MessageContext axis2MsgCtx =
+                ((org.apache.synapse.core.axis2.Axis2MessageContext) msgCtx).getAxis2MessageContext();
+        //setting transport headers
+        axis2MsgCtx.setProperty(MessageContext.TRANSPORT_HEADERS
+                , receivedEvent.getHeadersMap());
+        String contentType = receivedEvent.getHeadersMap().get(InboundGrpcConstants.HEADER_MAP_CONTENT_TYPE_PARAMETER_NAME);
+        if (log.isDebugEnabled()) {
+            log.debug(contentType + " Content-Type, received via the gRPC headers.");
+        }
+        // Determine the message builder to use
+        if (contentType != null) {
+            if (InboundGrpcConstants.CONTENT_TYPE_JSON.equalsIgnoreCase(contentType)) {
+                contentType = InboundGrpcConstants.CONTENT_TYPE_JSON_MIME_TYPE;
+            } else if (InboundGrpcConstants.CONTENT_TYPE_XML.equalsIgnoreCase(contentType)) {
+                contentType = InboundGrpcConstants.CONTENT_TYPE_XML_MIME_TYPE;
+            } else {
+                log.error("Error occurred when processing gRPC message. " + contentType +
+                        " type found in gRPC header is not supported");
+                responseObserver.onError(
+                        new Throwable("Error occurred when processing gRPC message. " + contentType +
+                                " type found in gRPC header is not supported"));
+                return;
+            }
+        } else {
+            log.error("Invalid content type found in gRPC header. JSON, XML and text is supported");
```

Comment:
We haven't handled test type above

Reply:
fixed in https://github.com/wso2/micro-integrator/pull/854/commits/fff5493dd60f14989271869d8d399bb396a56033